recipes/openssl/1.x.x/conanfile.py

```diff
@@ -791,8 +794,11 @@ def _patch_install_name(self):
 
     def _replace_runtime_in_file(self, filename):
         for e in ["MDd", "MTd", "MD", "MT"]:
-            tools.replace_in_file(filename, "/%s " % e, "/%s " % msvc_runtime_flag(self), strict=False)
-            tools.replace_in_file(filename, "/%s\"" % e, "/%s\"" % msvc_runtime_flag(self), strict=False)
+            flag = msvc_runtime_flag(self)
+            if self._is_clangcl:
+                flag = self.settings.get_safe("compiler.runtime")
```

Comment:
So `msvc_runtime_flag()` lacks clang-cl support. Would be nice to open an issue in conan client.

Reply:
totally!

Reply:
@SpaceIm https://github.com/conan-io/conan/pull/10898 😉

Reply:
Thanks, so I'm wondering if this modification in openssl is robust enough. It will break if user uses new model of `compiler.runtime`.

Reply:
it will need to add a support for new model, like `compiler.runtime` and `compiler.runtime_type`. we'll also have to change our `clang-cl` profiles.

Reply:
```suggestion
            flag = msvc_runtime_flag(self)
            if self._is_clangcl and not flag:
                flag = self.settings.get_safe("compiler.runtime")
```

I mean something like this.

Reply:
good point to make it robust. I thought, we will need to wait till new conan version is released and add a conan_version_required to the recipe

Reply:
Well actually https://github.com/conan-io/conan/pull/10898 raises ConanException if compiler is not Visual Studio and compiler.runtime is not None, static or dynamic, so it will break this PR anyway.